### PR TITLE
add Project Filter Page to some source

### PR DESCRIPTION
**Pull request by Riztard**
another way to open websiteUrl.com/project page
example: https://komikav.com/project/

# Source:
WPMangaStream
- Boosei
- KaisarKomik
- KlanKomik
- KomikAV
- KomikCast
- KomikIndo.co
- KomikStation
- KumaScans
- MasterKomik
- ReadKomik
- SekteKomik
- SheaManga
- WestManga
- WorldRomanceTranslation
- xCaliBRScans

WPMangaReader
- Kiryuu
- MangaKita
- Ngomik

Madara
- PojokManga

Others
- Komiku
- MaidManga
- MangaKane
- ManhuaID

# Screenshoot Filter
image
image

### CODE REVIEW

**Comment by Riztard** [comment]
hmm

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/wpmangareader/WPMangaReader.kt [hunk 1]
```diff
@@ -274,15 +280,45 @@ abstract class WPMangaReader(
...
+                Filter.Header("NOTE: Ignored if using text search!"),
+                GenreFilter(),
+                StatusFilter(),
+                TypesFilter(),
+                OrderByFilter(),
+                Filter.Separator(),
+                Filter.Header("NOTE: cant be used with other filter!"),
+                Filter.Header("$name Project List page"),
+                ProjectFilter(),
+            )
```
Comment (CarlosEsco) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/wpmangareader/WPMangaReader.kt:306]: Should be able to replace this with something like
```
override fun getFilterList(){
val filters = mutableListOf<Filter<*>>(
    Filter.Header("NOTE: Ignored if using text search!"),
    GenreFilter(),
    StatusFilter(),
    TypesFilter(),
    OrderByFilter()
)
if(hasProjectPage){
  filters.addAll(mutableListOf<Filter<*>(
      Filter.Separator(),
      Filter.Header("NOTE: cant be used with other filter!"),
      Filter.Header("$name Project List page"),
      ProjectFilter(),
  )
}
return FilterList(filters)
}

```

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/wpmangastream/WPMangaStream.kt [hunk 2]
```diff
@@ -352,20 +358,49 @@ abstract class WPMangaStream(
...
+                YearFilter(),
+                StatusFilter(),
+                TypeFilter(),
+                SortByFilter(),
+                GenreListFilter(getGenreList()),
+                Filter.Separator(),
+                Filter.Header("NOTE: cant be used with other filter!"),
+                Filter.Header("$name Project List page"),
+                ProjectFilter(),
+            )
```
Comment (CarlosEsco) [inline multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/wpmangastream/WPMangaStream.kt:402]: similar like above

File: multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/wpmangareader/WPMangaReader.kt (hunk 1 above)
Comment (Riztard) [reply multisrc/src/main/java/eu/kanade/tachiyomi/multisrc/wpmangareader/WPMangaReader.kt:306]: ah addAll, was wondering how to add it

**Comment by Riztard** [comment]
round 2 
READY

**Pull request merged**